File: master_session.m
vmax=1;  km=1;  p=[vmax;km];             % Simulation study. First set
x=(0:.1:2)';    y=zeros(21,1);           % p to known values, then
[s,r,f]=hyperb(p,x,y);                   % generate artificial data.
niter=50; stol=.001; erra=1;             % Request for error analysis.
y=f+.01*randn(21,1);                     % Perturb p.  See how well p
dpr=ones(2,1); dpa=dpr;                  % is recovered by fitting to
ponoff = [];
[ p,se,dep,  r,f,s,  iter,cnvrg,qi ]= ...      % the noisy data.  Note: x & y
    rsleasqr('hyperb','dhyperb', ...           % are arguments to be passed
    p,dpr,dpa,ponoff, ...                 % to hyperb & dhyperb.
    stol,niter,erra,x,y);
disp('Parameters Std. errs. Dependencies');
disp([p,se,dep]);                        % Show parameters et al.
figure
plot(x,y,'o')
hold on
plot(x,f,'k');                   % Plot y and fitted f.

File: master_session.r
rm(list = ls())

source_list = list(
  'dhyperb.r',
  'hyperb.r',
  'rsleasqr.r'
)

for (src in source_list) {
  source(file.path(getwd(), src))
}


# set up some dummy values
vmax <- 1;
km   <- 1;
p    <- t(c(vmax, km))
x    <- seq(0, 2, by=0.1)
y    <- matrix(0, 21, 1)

# generate some fake data
out <- hyperb(p, x, y)
ss  <- out$ss
r   <- out$r
ff  <- out$ff

# request for error analysis
niter <- 50
stol  <- 0.001
erra  <- 1;

# make some noise
y      <- ff+0.01 * rnorm(21)
dpr    <- matrix(1, 2, 1)
dpa    <- dpr
ponoff <- NULL

# run it
out <- rsleasqr('hyperb', 'dhyperb', p, dpr, dpa, ponoff, stol, niter, erra, x, y)

# display results
print('Parameters  Std err  Dependencies')
print(matrix(c(out$pbest, out$se, out$dep),2))
plot(x,y)
matplot(x,out$fbest,'l',add=T)
